optimum/exporters/onnx/utils.py

```diff
@@ -199,3 +199,28 @@ def recursive_to_device(value: Union[Tuple, List, "torch.Tensor"], device: str):
         value = value.to(device)
 
     return value
+
+
+def recursive_to_dtype(value: Union[Tuple, List, "torch.Tensor"], dtype: Optional[torch.dtype]):
+    if dtype is None:
+        return value
+
+    if isinstance(value, tuple):
```

Comment:
Why do you distinguish between tuple and list, the difference between them and the last case are to check if `value` is iterable no?

Reply:
Tuple does not support item assignment.